src/components/_common/Input/Input.tsx

```diff
@@ -0,0 +1,212 @@
+import styled from "@emotion/styled";
+import { useState, KeyboardEvent } from "react";
+import InputUpload from "./InputUpload";
+
+interface InputProps {
+  width?: string;
+  height?: string;
+  placeholder: string;
+  isTextarea?: boolean;
+  hasTags?: boolean;
+  tags?: string[];
+  hasImage?: boolean;
+  image?: string;
+}
+
+interface IInputStyle{
+  width: string;
+  height: string;
+}
+
+export const Input = ({
+  width = '100%',
+  height = 'auto',
+  placeholder,
+  isTextarea = false,
+  hasTags = false,
+  tags: initialTags = [],
+  hasImage = false,
+  image: initialImage,
+}: InputProps) => {
```

Comment:
이건 개인적인 의견입니다!

지금 굳이 이걸 할 필요는 없는데 이후에 인적오류를 방지하기위해 반영되면 좋을 것 같습니다!

```suggestion
export const Input = ({
  width = '100%',
  height = 'auto',
  placeholder,
  isTextarea = false,
  tags: initialTags = [],
  image: initialImage,
}: InputProps) => {
```

`hasTags`와 `hasImage`를 현재 컴포넌트에서 판단할 수 있습니다!
```js
const hasTags = Array.isArray(initialTags) && initialTags.length > 0;
const hasImage = initialImage != null    // undefined까지 포함해서 걸러줍니다!
```

Reply:
말씀대로 인지를 하고 있었습니다.
제가 인지했던 문제 사항은 다시 값을 들고 올 경우(포스트 수정) 였습니다.
쓸데없이 너무 많이 생각했던 것 같네요..
차라리 말씀대로 진행을 하고 image: '', 이런식으로 빈 값이라도 던져줘서 컨트롤 하는게 나을 것 같습니다.
감사합니다~

Reply:
제가 잘 이해를 못했을 수도 있는데

포스트 수정을 처리하는 곳에서 input에 자기가 가지고 있는 

포스트 정보(글, 이미지)를 넘겨주면 괜찮지 않을까요?

일단 지금 당장 input을 쓰는데는 지장이 없을 것 같아 보이는데

예외케이스가 생기면 수정하는 식으로 대응해야 할 것 같아요